packages/integrations/sitemap/src/index.ts

```diff
@@ -168,13 +168,15 @@ const createPlugin = (options?: SitemapOptions): AstroIntegration => {
 						}
 					}
 					const destDir = fileURLToPath(dir);
+					const xslURL = opts.xslURL ? new URL(opts.xslURL, finalSiteUrl).href : undefined;
```

Comment:
I am not really sure how zod does the validation for URLs, but I suppose it wouldn't hurt to use [`URL.canParse`](https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static), and make sure we don't get runtime errors.

Reply:
Zod should already valid this https://github.com/colinhacks/zod/blob/7173d0bcc2105777102e22d36a2866196e2830f3/src/types.ts#L819-L821